### ISSUE TITLE
Add a note in the migration guide on Flyway#repair

### ISSUE THREAD

**Issue by dsyer** [spring-projects/spring-boot#17738]
Just running the same app with Spring Boot 1.5 and Flyway 4.2.0 does not fix (or change) the checksums. This does appear to work:

```java
    @Bean
    public FlywayMigrationStrategy repairStrategy() {
        return flyway -> {
            flyway.repair();
            flyway.migrate();
        };
    }
```
Also probably worth mentioning that it might help to set `spring.flyway.table=schema_version` (because they changed the default table name, and Flyway will break in the future if you don't do that).

There is also a Maven plugin with a `repair` goal, so that might work too. Not sure about Gradle.

**Comment by snicoll** [comment]
As you've been experiencing that with a particular project, could it be that the `repair` part is something specific to it? I don’t know if we can upgrade the migration guide for all Spring Boot users without knowing more about what can be wrong in the current guide.

We could certainly mention it but getting a bit more details should help (what was the stacktrace, etc)

**Comment by dsyer** [comment]
I have no idea to be honest what `flyway.repair()` does. But it's a feature of Flyway, so I don't know why it's there if it is specific to the application I was working on.  Example stack trace fragment below:

```
Caused by: org.flywaydb.core.api.FlywayException: Validate failed. Migration Checksum mismatch for migration 2
-> Applied to database : 1499248173
-> Resolved locally    : -1729781252
    at org.flywaydb.core.Flyway.doValidate(Flyway.java:1108) ~[flyway-core-5.2.4.jar:na]
    at org.flywaydb.core.Flyway.access$300(Flyway.java:62) ~[flyway-core-5.2.4.jar:na]
    at org.flywaydb.core.Flyway$1.execute(Flyway.java:1012) ~[flyway-core-5.2.4.jar:na]
    at org.flywaydb.core.Flyway$1.execute(Flyway.java:1006) ~[flyway-core-5.2.4.jar:na]
    at org.flywaydb.core.Flyway.execute(Flyway.java:1418) ~[flyway-core-5.2.4.jar:na]
    at org.flywaydb.core.Flyway.migrate(Flyway.java:1006) ~[flyway-core-5.2.4.jar:na]
    at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66) ~[...]
```

**Comment by snicoll** [comment]
I've added [a note in the relevant section](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#flyway). I didn't mention the table change as there is a warning and users should refer to the Flyway documentation for that anyway.
